### PR TITLE
Deploy cronjobs in OCP4 dev

### PR DESCRIPTION
**Pull request by andrea-williams**

### CODE REVIEW

**Review by Sybrand — COMMENTED** [review]
Please at OCP4 Jenkins deploy and test.

File: openshift/templates/bcfw_p1_forecasts.cronjob.yaml [hunk 1]
```diff
@@ -15,13 +15,19 @@ parameters:
...
+    value: docker-registry.default.svc:5000
   - name: JOB_NAME
     value: bcfw-p1-forecasts
   - name: SCHEDULE
     value: "30 8,16 * * *"
     required: true
+  - name: POSTGRES_WRITE_HOST
+    required: true
+  - name: POSTGRES_USER
+  - name: POSTGRES_DATABASE
```
Comment (Kyubinhan) [inline openshift/templates/bcfw_p1_forecasts.cronjob.yaml:30]: I think these should have default values or need to be required?

File: openshift/templates/bcfw_p1_hourly_actuals.cronjob.yaml [hunk 2]
```diff
@@ -15,13 +15,18 @@ parameters:
...
+    value: docker-registry.default.svc:5000
   - name: JOB_NAME
     value: bcfw-p1-hourly-actuals
   - name: SCHEDULE
-    value: "15 * * * *"
     required: true
+  - name: POSTGRES_WRITE_HOST
+    required: true
+  - name: POSTGRES_USER
+  - name: POSTGRES_DATABASE
```
Comment (Kyubinhan) [inline openshift/templates/bcfw_p1_hourly_actuals.cronjob.yaml:29]: Same here

File: openshift/templates/env_canada_gdps.cronjob.yaml [hunk 3]
```diff
@@ -15,10 +15,22 @@ parameters:
...
+  - name: PROJ_TOOLS
     value: auzhsi-tools
   - name: JOB_NAME
     value: env-canada
+  - name: IMAGE_REGISTRY
+    required: true
+    value: docker-registry.default.svc:5000
+  - name: POSTGRES_WRITE_HOST
+    required: true
+  - name: POSTGRES_READ_HOST
```
Comment (Kyubinhan) [inline openshift/templates/env_canada_gdps.cronjob.yaml:27]: Hey sorry looks like this one also needs to be required

File: openshift/templates/env_canada_rdps.cronjob.yaml [hunk 4]
```diff
@@ -15,10 +15,22 @@ parameters:
...
+  - name: PROJ_TOOLS
     value: auzhsi-tools
   - name: JOB_NAME
     value: env-canada
+  - name: IMAGE_REGISTRY
+    required: true
+    value: docker-registry.default.svc:5000
+  - name: POSTGRES_WRITE_HOST
+    required: true
+  - name: POSTGRES_READ_HOST
```
Comment (Kyubinhan) [inline openshift/templates/env_canada_rdps.cronjob.yaml:27]: same here

File: openshift/templates/env_canada_hrdps.cronjob.yaml [hunk 5]
```diff
@@ -15,10 +15,22 @@ parameters:
...
+  - name: PROJ_TOOLS
     value: auzhsi-tools
   - name: JOB_NAME
     value: env-canada
+  - name: IMAGE_REGISTRY
+    required: true
+    value: docker-registry.default.svc:5000
+  - name: POSTGRES_WRITE_HOST
+    required: true
+  - name: POSTGRES_READ_HOST
```
Comment (Kyubinhan) [inline openshift/templates/env_canada_hrdps.cronjob.yaml:27]: same here

File: openshift/Jenkinsfile.ocp4 [hunk 6]
```diff
@@ -47,12 +47,12 @@ pipeline {
                     sh "CPU_REQUEST=500m CPU_LIMIT=500m MEMORY_REQUEST=2Gi MEMORY_LIMIT=2Gi REPLICAS=3 PROJ_TARGET=${projProd} PROJ_TOOLS=e1e498-tools POSTGRES_USER=wps POSTGRES_DATABASE=wps IMAGE_REGISTRY=image-registry.openshift-image-registry.svc:5000 SECOND_LEVEL_DOMAIN=apps.silver.devops.gov.bc.ca POSTGRES_WRITE_HOST=patroni-wps-prod-leader POSTGRES_READ_HOST=patroni-wps-prod-replica ./openshift/scripts/oc_deploy.sh prod apply"
                     // TODO: Apply these additional steps:
                     // // Env Canada Subscriber
```
Comment (Sybrand) [inline openshift/Jenkinsfile.ocp4:49]: Pedantic, but can you drop the double //?
```suggestion
                    // Env Canada Subscriber
```

File: openshift/Jenkinsfile.ocp4 [hunk 7]
```diff
@@ -47,12 +47,12 @@ pipeline {
                     sh "CPU_REQUEST=500m CPU_LIMIT=500m MEMORY_REQUEST=2Gi MEMORY_LIMIT=2Gi REPLICAS=3 PROJ_TARGET=${projProd} PROJ_TOOLS=e1e498-tools POSTGRES_USER=wps POSTGRES_DATABASE=wps IMAGE_REGISTRY=image-registry.openshift-image-registry.svc:5000 SECOND_LEVEL_DOMAIN=apps.silver.devops.gov.bc.ca POSTGRES_WRITE_HOST=patroni-wps-prod-leader POSTGRES_READ_HOST=patroni-wps-prod-replica ./openshift/scripts/oc_deploy.sh prod apply"
                     // TODO: Apply these additional steps:
```
Comment (Sybrand) [inline openshift/Jenkinsfile.ocp4:48]: Pls. move the TODO down to backups.

File: openshift/Jenkinsfile.ocp4 [hunk 8]
```diff
@@ -47,12 +47,12 @@ pipeline {
...
-                    // sh "PROJ_TARGET=${projProd} PROJ_TOOLS=e1e498-tools ./openshift/scripts/oc_provision_ec_rdps_cronjob.sh prod apply"
+                    sh "PROJ_TARGET=${projProd} PROJ_TOOLS=e1e498-tools IMAGE_REGISTRY=image-registry.openshift-image-registry.svc:5000 POSTGRES_WRITE_HOST=patroni-wps-prod-leader POSTGRES_READ_HOST=patroni-wps-prod-replica POSTGRES_DATABASE=wps POSTGRES_USER=wps ./openshift/scripts/oc_provision_ec_gdps_cronjob.sh prod apply"
+                    sh "PROJ_TARGET=${projProd} PROJ_TOOLS=e1e498-tools IMAGE_REGISTRY=image-registry.openshift-image-registry.svc:5000 POSTGRES_WRITE_HOST=patroni-wps-prod-leader POSTGRES_READ_HOST=patroni-wps-prod-replica POSTGRES_DATABASE=wps POSTGRES_USER=wps ./openshift/scripts/oc_provision_ec_hrdps_cronjob.sh prod apply"
+                    sh "PROJ_TARGET=${projProd} PROJ_TOOLS=e1e498-tools IMAGE_REGISTRY=image-registry.openshift-image-registry.svc:5000 POSTGRES_WRITE_HOST=patroni-wps-prod-leader POSTGRES_READ_HOST=patroni-wps-prod-replica POSTGRES_DATABASE=wps POSTGRES_USER=wps ./openshift/scripts/oc_provision_ec_rdps_cronjob.sh prod apply"
                     // // BC FireWeather cronjobs
```
Comment (Sybrand) [inline openshift/Jenkinsfile.ocp4:53]:
```suggestion
                    // BC FireWeather cronjobs
```

File: openshift/templates/bcfw_p1_hourly_actuals.cronjob.yaml [hunk 9]
```diff
@@ -41,7 +50,7 @@ objects:
             spec:
               containers:
                 - name: ${JOB_NAME}
-                  image: docker-registry.default.svc:5000/${PROJECT_TOOLS}/${NAME}-${SUFFIX}:${SUFFIX}
+                  image: ${IMAGE_REGISTRY}/${PROJ_TOOLS}/${NAME}-${SUFFIX}:${SUFFIX}
```
Comment (Sybrand) [inline openshift/templates/bcfw_p1_hourly_actuals.cronjob.yaml:53]: @andrea-williams have you re-run jenkins in prod on silver? It's failing there, still pulling from the old stuff.

**Review by Sybrand — COMMENTED** [review]
Looking good - just not working in silver yet.

**Review by Sybrand — APPROVED** [review]

**Review by Kyubinhan — APPROVED** [review]

**Pull request merged**